packages/base/lua/Profiler.lua

```diff
@@ -0,0 +1,310 @@
+		end
+
+		return block.count
+	end,
+	--- Return how much time was spent on the block.
+	-- It returns two representations: a string with a human-like representation of the time
+	-- and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it returns the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- Profiler():stop("block")
+	-- stringTime, numberTime = Profiler():uptime("block")
+	uptime = function(self, name)
+		optionalArgument("name", "string", name)
+		local block = self.blocks[name or self:current().name]
+		if not block then
+			customError(string.format("Block '%s' was not found.", name))
+		end
+
+		local time = block:uptime()
+		return timeToString(time), time
+	end,
+	--- Stop to measure the time of a given block. It also returns how much time was spent with the block
+	-- in two representations: a string with a human-like representation of the time and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it stops and return the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- stringTime, numberTime = Profiler():stop("block")
+	stop = function(self, name)
+		optionalArgument("name", "string", name)
+		if name == "main" or (not name and self:current() and self:current().name == "main") then
+			customWarning("The block 'main' cannot be stopped.")
+			return self:uptime(name)
+		end
+
+		local block
+		if not name then
+			block = Stack.pop(self.stack)
+		elseif self.blocks[name] then
+			block = self.blocks[name]
+			Stack.remove(self.stack, block)
+		else
+			customError(string.format("Block '%s' was not found.", name))
+		end
+
+		if block.running then
+			block:stop()
+		end
+
+		local time = block:uptime()
+		return timeToString(time), time
+	end,
+	--- Clean the Profiler, removing all blocks and restarting its execution time.
+	-- @usage Profiler():clean()
+	clean = function(self)
+		self.blocks = {}
+		self.stack = {}
+		self:start("main")
+	end,
+	--- Show a report with the time and amount of times each block was executed.
+	-- @usage Profiler():report()
+	report = function(self)
+		print(string.format("%-30s%-20s%-30s%s", "Block", "Count", "Time", "Average")) -- SKIP
+		forEachOrderedElement(self.blocks, function(_, block)
+			local report = block:report() -- SKIP
+			print(string.format("%-30s%-20d%-30s%s", report.name, report.count, report.time, report.average)) -- SKIP
+		end)
+
+		local _, total = self:uptime("main")
+		print("Total execution time: "..timeToString(total)) -- SKIP
+	end,
+	--- Define how many times a given block will be executed.
+	-- @arg name A string with the block name.
+	-- @arg quantity Number of steps a given block will execute.
+	-- @usage Profiler():start("block")
+	-- Profiler():steps("block", 5)
+	-- Profiler():stop("block")
+	steps = function(self, name, quantity)
+		mandatoryArgument("name", "string", name)
```

Comment:
The user does not know that the first argument is named `name`. She only knows that it is the first argument. Therefore use:
```lua
mandatoryArgument(1, "string", name)
```
Do the same for the other arguments when the function does not get a named table as argument.